test/Microsoft.AspNet.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -687,96 +687,154 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
             Assert.Same(stringCollectionWriter.Buffer.BufferEntries, buffer.BufferEntries[1]);
         }
 
-        [Fact]
-        public async Task Write_ITextWriterCopyable_WritesContent()
+        public static TheoryData<TagHelperOutput, string> WriteTagHelper_InputData
+        {
+            get
+            {
+                // parameters: TagHelperOutput, expectedOutput
+                return new TheoryData<TagHelperOutput, string>
+                {
+                    {
+                        // parameters: TagName, Attributes, SelfClosing, PreContent, Content, PostContent
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), false, null, "Hello World!", null),
+                        "<p>Hello World!</p>"
+                    },
+                    {
+                        GetTagHelperOutput(null, new Dictionary<string, string>(), false, null, "Hello World!", null),
+                        "Hello World!"
+                    },
+                    {
+                        GetTagHelperOutput(
+                            "p",
+                            new Dictionary<string, string>() { { "test", "testVal" } },
+                            false,
+                            null,
+                            "Hello World!",
+                            null),
+                        "<p test=\"testVal\">Hello World!</p>"
+                    },
+                    {
+                        GetTagHelperOutput(
+                            "p",
+                            new Dictionary<string, string>() { { "test", "testVal" } },
+                            true,
+                            null,
+                            "Hello World!",
+                            null),
+                        "<p test=\"testVal\" />"
+                    },
+                    {
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), false, "Hello World!", null, null),
+                        "<p>Hello World!</p>"
+                    },
+                    {
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), false, null, null, "Hello World!"),
+                        "<p>Hello World!</p>"
+                    },
+                    {
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), false, "Hello World!", null, null),
+                        "<p>Hello World!</p>"
+                    },
+                    {
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), false, "Hello", "Test", "World!"),
+                        "<p>HelloTestWorld!</p>"
+                    },
+                    {
+                        GetTagHelperOutput("p", new Dictionary<string, string>(), true, "Hello", "Test", "World!"),
+                        "<p />"
+                    }
```

Comment:
You'll need 1 test case for each test removed in `TagHelperOutputTest.cs`. Don't want to miss anything.